todo_app/templates/index.html

```diff
@@ -7,17 +7,23 @@ <h1 class="display-4">To-Do App</h1>
     <p class="lead">Just another to-do app.</p>
   </div>
 
+  <form action="" method="get" class="form-example">
```

Comment:
This is sending a GET request to the base url. i.e. it is getting routed to the `index` function, rather than the `add` function.

Fix the `action` and `method`.